### PR TITLE
Update to kotlin 1.8.0

### PR DESCRIPTION
**Pull request by eymar**
Also:
- update skiko version
- cherry-pick js and native related fixes for compose compiler plugin

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt [hunk 1]
```diff
@@ -68,7 +68,7 @@ internal class FlushCoroutineDispatcher(
      * Does the dispatcher have any tasks scheduled or currently in progress
      */
     fun hasTasks() = synchronized(tasksLock) {
-        tasks.isNotEmpty() && delayedTasks.isNotEmpty()
+        tasks.isNotEmpty() || delayedTasks.isNotEmpty()
```
Comment (eymar) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt]: Important implementation change:
we had a wrong condition (not related to 1.8.0 update though, I can submit this change as a separete PR).

Comment (eymar) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt]: I removed this change from this PR. Here is a new PR with this change - https://github.com/JetBrains/androidx/pull/370

File: buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt [hunk 2]
```diff
@@ -390,7 +390,13 @@ private fun configureComposeCompilerPlugin(
 
             // Gradle hack ahead, we use of absolute paths, but is OK here because we do it in
             // doFirst which happens after Gradle task input snapshotting. AGP does the same.
-            compile.doFirst {
+
```
Comment (igordmn) [inline buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt:393]: I think, we shouldn't squash this PR, and preserve commits which were cherry-picked. We have 2 ways:

1. cherry-pick them to jb-main, and just push it without PR
2. allow rebase PR-merging

What do you think is better?

If we go p.2, better to mention which commits were cherry-picked, to make review easier (so we don't look at things that are already reviewed in AOSP)

Comment (eymar) [reply buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt:393]: p1 should be easier for review I think. I'll update the PR.

Comment (eymar) [reply buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt:393]: only 3 commits left now

**Review by igordmn — APPROVED** [review]

**Pull request merged**